### DISCUSSION TITLE
Obtaining devices current location on mobile platforms

### DISCUSSION THREAD

**Discussion by Asimovcoviz** [AvaloniaUI/Avalonia#11988] [Q&A] [answered]
Is there any way to get the current location of a device if the application is build on Android or iOS?
For example, on Android there is the Play Services Location Task that keeps track of the devices location, also there is a permission to access geolocation data.

My question is only how to get this information. Is there any way implemented?

**Comment by emmauss** [comment, marked as answer]
This is outside the scope of Avalonia. You can handle querying location in the android project, just like you would in a native android app, or use the Essentials package

**Reply by Asimovcoviz** [reply]
That was a hint to the right direction, thank you! I managed to make it work by including Xamarin.Essentials to my project.
I also found Avalonia.Essentials (which seems to be unfinished I guess) and Maui.Essentials, which only produced a bunch of creative errors.

So, if anyone is looking for an answer to this question: Simply include Xamarin.Essentials and you'll have access to all hardware related featuers.
